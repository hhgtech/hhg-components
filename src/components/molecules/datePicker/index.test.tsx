import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ReactTestUtils, { act } from 'react-dom/test-utils'
import 'jest-styled-components'

import { DatePicker } from '.'

import MockTheme from './../../../mockTheme'

afterEach(cleanup)

const placeholder = 'text holder'

test('it renders and calendar appears on click', () => {
  const datePicker = render(
    <MockTheme>
      <DatePicker
        placeholder={placeholder}
        displayDateFormat="dd-MM-yyyy"
        pickedDate={new Date()}
        customInput={(props: any) => <input {...props} />}
        onChange={(d: Date) => {
          console.log(d)
        }}
      />
    </MockTheme>,
  )

  const self = datePicker.container.firstChild
  const input = datePicker.container.querySelector('input')

  expect(self).toMatchSnapshot()
  expect(input.placeholder).toEqual(placeholder)

  let calendar = datePicker.container.querySelector('.pickerOverlayWrapper')

  expect(calendar).toBeNull()

  act(() => {
    input.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })

  calendar = datePicker.container.querySelector('.pickerOverlayWrapper')
  expect(calendar).toBeTruthy()
})
