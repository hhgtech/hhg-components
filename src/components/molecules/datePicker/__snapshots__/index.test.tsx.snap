// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and calendar appears on click 1`] = `
.c0 .pickerContainer {
  position: relative;
  padding-top: 0;
}

.c0 .pickerContainer > div > svg {
  position: absolute;
  top: 36px;
  right: 16px;
  cursor: pointer;
}

.c0 .pickerContainer > :not(:first-child) {
  position: absolute;
  z-index: 1;
  width: 100%;
}

.c0 .pickerOverlay {
  margin-top: 6px;
  background: white;
  border-radius: 6px;
  box-shadow: 0px 6px 12px rgba(67,86,100,0.2);
}

.c0 .picker {
  width: 100%;
}

.c0 .react-datepicker-popper .react-datepicker {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 340px;
  height: 305px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  box-shadow: (-1px 1px 2px rgba(67,70,74,0.0001)), (-2px 2px 5px rgba(67,86,100,0.123689));
}

.c0 .react-datepicker-popper .svg {
  cursor: pointer;
  stroke: #262626;
}

.c0 .react-datepicker-popper .react-datepicker__day {
  width: 40px;
}

.c0 .react-datepicker-popper .react-datepicker__header {
  border-bottom: 0;
  background-color: white;
}

.c0 .react-datepicker-popper .react-datepicker__day--outside-month {
  color: #e9e9e9;
}

.c0 .react-datepicker-popper .react-datepicker__day {
  font-weight: 700;
}

.c0 .react-datepicker-popper .react-datepicker__day--selected,
.c0 .react-datepicker-popper .react-datepicker__day--today,
.c0 .react-datepicker-popper .react-datepicker__day--selected:hover,
.c0 .react-datepicker-popper .react-datepicker__day--keyboard-selected {
  width: 32px;
  height: 30px;
  margin: 4px;
  margin-bottom: -2px;
  background: #2d87f3;
  border-radius: 50%;
}

.c0 .react-datepicker-popper .react-datepicker__day-name {
  width: 40px;
}

.c0 .react-datepicker-popper .react-datepicker__triangle {
  display: none;
}

.c0 .react-datepicker-popper .react-datepicker__header.react-datepicker__header--custom svg {
  stroke: #262626;
}

.c0 .react-datepicker-popper .react-datepicker__header.react-datepicker__header--custom select {
  background: #e4e8ec;
  text-align: center;
}

.c0 .react-datepicker-popper .react-datepicker__header.react-datepicker__header--custom select:not(:last-child) {
  padding: 8px 10px;
  margin-right: 15px;
  border-radius: 6px;
}

.c0 .DayPicker {
  display: inline-block;
  font-size: 1rem;
}

.c0 .DayPicker-wrapper {
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-bottom: 1em;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .DayPicker-Months {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .DayPicker-Month {
  display: table;
  margin: 0 1em;
  margin-top: 1em;
  border-collapse: collapse;
  border-spacing: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .DayPicker-NavButton {
  position: absolute;
  top: 35px;
  display: inline-block;
  width: 1.25em;
  height: 1.25em;
  margin-top: 2px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  color: #8b9898;
  cursor: pointer;
}

.c0 .DayPicker-NavButton:hover {
  opacity: 0.8;
}

.c0 .DayPicker-NavButton--prev {
  top: 24px;
  left: 2px;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='10' viewBox='0 0 6 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M5 9L1 5L5 1' stroke='%23262626' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
}

.c0 .DayPicker-NavButton--next {
  top: 24px;
  right: 2px;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='10' viewBox='0 0 6 10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1 1L5 5L1 9' stroke='%23262626' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
}

.c0 .DayPicker-NavButton--interactionDisabled {
  display: none;
}

.c0 .DayPicker-Caption {
  display: table-caption;
  width: 100%;
  text-align: center;
}

.c0 .DayPicker-Caption > div {
  font-size: 1.15em;
  font-weight: 500;
}

.c0 .DayPicker-Weekdays {
  display: table-header-group;
  margin-top: 1em;
}

.c0 .DayPicker-WeekdaysRow {
  display: table-row;
}

.c0 .DayPicker-Weekday {
  display: table-cell;
  padding: 0.5em;
  color: #8b9898;
  font-size: 0.875em;
  text-align: center;
}

.c0 .DayPicker-Weekday abbr[title] {
  border-bottom: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .DayPicker-Body {
  display: table-row-group;
}

.c0 .DayPicker-Week {
  display: table-row;
}

.c0 .DayPicker-Day {
  display: table-cell;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  cursor: pointer;
  text-align: center;
  vertical-align: middle;
}

.c0 .DayPicker-WeekNumber {
  display: table-cell;
  min-width: 1em;
  padding: 0.5em;
  border-right: 1px solid #eaecec;
  color: #8b9898;
  cursor: pointer;
  font-size: 0.75em;
  text-align: right;
  vertical-align: middle;
}

.c0 .DayPicker--interactionDisabled .DayPicker-Day {
  cursor: default;
}

.c0 .DayPicker-Footer {
  padding-top: 0.5em;
}

.c0 .DayPicker-TodayButton {
  border: none;
  background-color: transparent;
  background-image: none;
  box-shadow: none;
  color: #4a90e2;
  cursor: pointer;
  font-size: 0.875em;
}

.c0 .DayPicker-Day--today {
  color: #d0021b;
  font-weight: 700;
}

.c0 .DayPicker-Day--outside {
  color: #8b9898;
  cursor: default;
}

.c0 .DayPicker-Day--disabled {
  color: #dce0e0;
  cursor: default;
}

.c0 .DayPicker-Day--sunday {
  background-color: #f7f8f8;
}

.c0 .DayPicker-Day--sunday:not(.DayPicker-Day--today) {
  color: #dce0e0;
}

.c0 .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
  position: relative;
  background-color: #2d87f3;
  color: #f0f8ff;
  font-weight: 700;
}

.c0 .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside):hover {
  background-color: #51a0fa;
}

.c0 .DayPicker:not(.DayPicker--interactionDisabled) .DayPicker-Day:not(.DayPicker-Day--disabled):not(.DayPicker-Day--selected):not(.DayPicker-Day--outside):hover {
  background-color: #f0f8ff;
}

.c0 .DayPickerInput {
  display: inline-block;
}

.c0 .DayPickerInput-OverlayWrapper {
  position: relative;
}

.c0 .DayPickerInput-Overlay {
  position: absolute;
  z-index: 1;
  left: 0;
  background: white;
  box-shadow: 0 2px 5px rgba(0,0,0,0.15);
}

<div
  class="c0"
>
  <div
    class="pickerContainer"
  >
    <input
      placeholder="text holder"
      readonly=""
      value=""
    />
  </div>
</div>
`;
